jdk_17_maven/em/embedded/rest/signal-server/src/main/java/em/embedded/textsecuregcm/EmbeddedEvoMasterController.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    private WhisperServerService application;
+
+    @Override
+    public boolean isSutRunning() {
+        if (application == null) {
+            return false;
+        }
+
+        return application.getJettyServer().isRunning();
+    }
+
+    @Override
+    public String getPackagePrefixesToCover() {
+        return "org.whispersystems.textsecuregcm.";
+    }
+
+    @Override
+    public List<AuthenticationDto> getInfoForAuthentication() {
+        return null;
+    }
+
+    @Override
+    public ProblemInfo getProblemInfo() {
+        return new RestProblem(
+                "http://0.0.0.0:" + application.getJettyPort() + "/assets/swagger.json",
+                null
+        );
+    }
+
+    @Override
+    public SutInfoDto.OutputFormat getPreferredOutputFormat() {
+        return SutInfoDto.OutputFormat.JAVA_JUNIT_5;
+    }
+
+    @Override
+    public String startSut() {
+        System.setProperty("aws.region", "us-west-2");
+
+        redisContainer.start();
+
+        dynamoDBContainer.start();
+
+        application = new WhisperServerService();
+
+        String redisURL = "redis://" + redisContainer.getHost() + ":" + redisContainer.getMappedPort(REDIS_PORT) + "/";
+
+        //Dirty hack for DW...
+        System.setProperty("dw.server.applicationConnectors[0].port", "0");
+        System.setProperty("dw.server.adminConnectors[0].port", "0");
+
+        System.setProperty("dw.cacheCluster.configurationUri", redisURL);
+        System.setProperty("dw.clientPresenceCluster.configurationUri", redisURL);
+        System.setProperty("dw.pubsub.uri", redisURL);
+        System.setProperty("dw.pushSchedulerCluster.configurationUri", redisURL);
+        System.setProperty("dw.rateLimitersCluster.configurationUri", redisURL);
+        System.setProperty("dw.messageCache.cluster.configurationUri", redisURL);
+        System.setProperty("dw.metricsCluster.configurationUri", redisURL);
+
+        try {
+            application.run("server", "src/main/resources/em-sample.yml");
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+
+        try {
+            Thread.sleep(3_000);
+        } catch (InterruptedException e) {
+
+        }
+
+        while(!application.getJettyServer().isStarted()) {
+            try {
+                Thread.sleep(3_000);
+            } catch (InterruptedException e) {
+
+            }
+        }
+
+        return "http://localhost:" + application.getJettyPort();
+    }
+
+    @Override
+    public void stopSut() {
+        if (application != null) {
+            try {
+                application.getJettyServer().stop();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        redisContainer.stop();
+        dynamoDBContainer.stop();
+    }
+
+    @Override
+    public void resetStateOfSUT() {
+        JedisPool jedisPool = new JedisPool("0.0.0.0", redisContainer.getFirstMappedPort());
```

Comment:
you can get the hostname/ip from `redisContainer`. also, such object should rather be at class instance, so it is not recreated at each call of this method